assets/src/dashboard/app/router/routerProvider.js

```diff
@@ -46,11 +46,15 @@ function RouterProvider({ children, ...props }) {
     });
   }, []);
 
-  useEffect(() => {
+  const currentScreen = useMemo(() => {
     const query = stringify(queryParams);
-    trackScreenView(currentPath + (query ? `?${query}` : ''));
+    return currentPath + (query ? `?${query}` : '');
   }, [currentPath, queryParams]);
 
+  useEffect(() => {
+    trackScreenView(currentScreen);
+  }, [currentScreen]);
+
```

Comment:
What's the difference here?

Reply:
Before that, the `trackScreenView` function was called multiple times when switching between screens (e.g. when going from "My Stories" to "Explore Templates"). Now it only happens once.

Reply:
Ah, if the `currentScreen` variable didn't change, even though the underlying parts did, it would trigger again.